### DISCUSSION TITLE
03-pytorch_computer_vision: question about calculating test accuracy (timestamp = 16:08:10)

### DISCUSSION THREAD

**Discussion by davidfelipeb** [mrdbourke/pytorch-deep-learning#152] [Q&A]
Hello! Thanks for this videos, they are really helpful! I have a question regarding the calculation of test accuracy in the `FashionMNISTModelV0`. In the video (see exact timestamp in title), you say that we should do 

> `test_acc += accuracy_fn(y_true = y_test, y_pred = test_pred.argmax(dim = 1))`

in order to correctly calculate the accuracy, that is because the variable `test_pred` contains logits, and not label predictions, and hence we should apply the `argmax` values on them. My question is: shouldn't we first convert  `test_pred` from logits to prediction probabilities and _then_ use argmax to find the maximum value? Something like this: 

> `test_acc += accuracy_fn(y_true = y_test, y_pred = torch.softmax(test_pred, dim = 1).argmax(dim = 1))`

Or am I missing something here?  Thanks for any help provided!

**Comment by fivefishstudios** [comment]
I specifically came here to ask the same question. 

Looking at output format of y_pred=Model(X), it seems softmax() and argmax() need to be applied first? y_pred and y are definitely not the same format.

**Comment by mrdbourke** [comment]
Hey @davidfelipeb,

Yes, you're right that is a valid option: logits -> prediction probabilities (`torch.softmax`) -> labels (`torch.argmax`).

But you can actually skip the middle step: logits -> labels (`torch.argmax`) - if you don't *require* prediction probabilities.

Confusing, yes. 

The first option is probably *more* right. The second option is more of a shortcut.

However, you'll notice they both give the same results.

You could try experimenting using it and not using it and seeing what happens.
